library/core/src/iter/traits/iterator.rs

```diff
@@ -58,6 +58,10 @@ fn _assert_is_object_safe(_: &dyn Iterator<Item = ()>) {}
         note = "if you want to iterate between `start` until a value `end`, use the exclusive range \
               syntax `start..end` or the inclusive range syntax `start..=end`"
     ),
+    on(
+        _Self = "{float}",
+        note = "if you want to create a range, the syntax is `start..end`"
```

Comment:
I'd recommend copying the text from above.

```suggestion
        note = "if you want to iterate between `start` until a value `end`, use the exclusive range \
              syntax `start..end` or the inclusive range syntax `start..=end`"
```

It *names* the syntax. This might be the very first error message someone sees when trying to use a for loop; they might not be familiar with `..` or know that it's called a "range". If they want to learn more about it, it's easy to look it up knowing what it's called.

This is a bit more subjective, but it also feels more like what the compiler would say 🦀

Reply:
Thanks.